tests/integration/test_specfile.py

```diff
@@ -684,8 +681,35 @@ def test_update_version(
     assert spec.expanded_version == version
 
 
-def test_trailing_newline(spec_autosetup, spec_no_trailing_newline):
-    spec = Specfile(spec_autosetup)
+def test_trailing_newline(specfile_factory, spec_autosetup, spec_no_trailing_newline):
+    spec = specfile_factory(spec_autosetup)
     assert str(spec)[-1] == "\n"
-    spec = Specfile(spec_no_trailing_newline)
+    spec = specfile_factory(spec_no_trailing_newline)
     assert str(spec)[-1] != "\n"
+
+
+def test_reload(specfile_factory, spec_minimal, spec_traditional):
+    spec = specfile_factory(spec_minimal)
+    before_reload = copy.deepcopy(spec)
+
+    if spec.path is not None:
+        os.remove(spec_minimal)
```

Comment:
You could make this parametrized with e.g. `remove_spec` argument that would be first `False` and then `True`, to test both cases.